demos/swipe_detection/swipe_detection_network.py

```diff
@@ -0,0 +1,266 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+# See: https://spdx.org/licenses/
+
+import typing as ty
+import numpy as np
+from multiprocessing import Pipe
+from lava.magma.compiler.compiler import Compiler
+from lava.magma.compiler.executable import Executable
+from lava.magma.core.process.message_interface_enum import ActorType
+from lava.magma.core.run_configs import Loihi2SimCfg, Loihi2HwCfg
+from lava.magma.core.run_conditions import RunSteps
+from lava.magma.runtime.runtime import Runtime
+from lava.lib.peripherals.dvs.prophesee import PropheseeCamera, PyPropheseeCameraRawReaderModel
+from lava.lib.peripherals.dvs.transformation import Compose, MergePolarities, \
+    Downsample
+from lava.proc.lif.process import LIF
+from lava.proc.dense.process import Dense
+from lava.proc.sparse.process import Sparse
+from lava.proc.dense.models import PyDenseModelBitAcc
+from lava.proc.lif.models import PyLifModelBitAcc
+from lava.proc.sparse.models import PySparseModelBitAcc
+from lava.proc.embedded_io.spike import NxToPyAdapter, PyToNxAdapter
+from lava.utils.serialization import save
+from outprocess.process_out import ProcessOut
+from scipy.sparse import csr_matrix
+from lava.magma.compiler.subcompilers.nc.ncproc_compiler import CompilerOptions
+
+CompilerOptions.verbose = True
+
+prophesee_input_default_config = {
+    "filename": "hand_swipe.dat",
+    "transformations": Compose(
+        [
+            Downsample(factor=0.125),
+            MergePolarities(),
+        ]
+    ),
+    "sensor_shape": (480, 640),
+    "num_output_time_bins": 1,
+    "sync_time": False,
+    "flatten": True
+}
+
+lif_default_config = {
+    "vth": 600,
+    "du": 2000,
+    "dv": 2000
+}
+
+weights_default_config = {
+    "w_ff": 255,
+    "w_rec": 5,
+    "w_o": 16,
+    "kernel_size": 20
+}
+
+num_out_neurons_default = 50
+blocking_default_config = False
+
+
+class SwipeDetector:
+    """Class to setup the swipe detection network, compile it, and initialize
+         its runtime."""
+
+    def __init__(self,
+                 send_pipe: type(Pipe),
+                 num_steps: int,
```

Comment:
Docstring for the input parameters would be nice, since this is the main class

Reply:
Done